SimMuon/MCTruth/src/MuonAssociatorByHits.cc

```diff
@@ -137,1349 +46,77 @@ MuonAssociatorByHits::associateRecoToSim( const edm::RefToBaseVector<reco::Track
-      int tpindex =0;
-      for (TrackingParticleCollection::const_iterator trpart = tPC.begin(); trpart != tPC.end(); ++trpart, ++tpindex) {
-
-	//	int n_tracker_simhits = 0;
-	int n_tracker_recounted_simhits = 0; 
-	int n_muon_simhits = 0; 
-	int n_global_simhits = 0; 
-	//	std::vector<PSimHit> tphits;
-
-	int n_tracker_selected_simhits = 0;
-	int n_muon_selected_simhits = 0; 
-	int n_global_selected_simhits = 0; 
-
-	// shared hits are counted over the selected ones
-	tracker_nshared = getShared(tracker_matchedIds_valid, trpart);
-	muon_nshared = getShared(muon_matchedIds_valid, trpart);
-
-        if (includeZeroHitMuons && n_muon_valid==0 && n_muon_INVALID!=0)
-	  muon_nshared = getShared(muon_matchedIds_INVALID, trpart);
-	
-        global_nshared = tracker_nshared + muon_nshared;	
-        if (global_nshared == 0) continue; // if this TP shares no hits with the current reco::Track loop over 
-
-	// This does not work with the new TP interface 
-	/*
-	for(std::vector<PSimHit>::const_iterator TPhit = trpart->pSimHit_begin(); TPhit != trpart->pSimHit_end(); TPhit++) {
-          DetId dId = DetId(TPhit->detUnitId());
-	  DetId::Detector detector = dId.det();
-	  
-	  if (detector == DetId::Tracker) {
-	    n_tracker_simhits++;
-
-	    unsigned int subdetId = static_cast<unsigned int>(dId.subdetId());
-	    if (!UsePixels && (subdetId==PixelSubdetector::PixelBarrel || subdetId==PixelSubdetector::PixelEndcap) )
-	      continue;
-
-	    SiStripDetId* stripDetId = 0;
-	    if (subdetId==SiStripDetId::TIB||subdetId==SiStripDetId::TOB||
-		subdetId==SiStripDetId::TID||subdetId==SiStripDetId::TEC)
-	      stripDetId= new SiStripDetId(dId);
-	    
-	    bool newhit = true;
-	    for(std::vector<PSimHit>::const_iterator TPhitOK = tphits.begin(); TPhitOK != tphits.end(); TPhitOK++) {
-	      DetId dIdOK = DetId(TPhitOK->detUnitId());
-	      //no grouped, no splitting
-	      if (!UseGrouped && !UseSplitting)
-		if (tTopo->layer(dId)==tTopo->layer(dIdOK) &&
-		    dId.subdetId()==dIdOK.subdetId()) newhit = false;
-	      //no grouped, splitting
-	      if (!UseGrouped && UseSplitting)
-		if (tTopo->layer(dId)==tTopo->layer(dIdOK) &&
-		    dId.subdetId()==dIdOK.subdetId() &&
-		    (stripDetId==0 || stripDetId->partnerDetId()!=dIdOK.rawId()))
-		  newhit = false;
-	      //grouped, no splitting
-	      if (UseGrouped && !UseSplitting)
-		if ( tTopo->layer(dId)== tTopo->layer(dIdOK) &&
-		    dId.subdetId()==dIdOK.subdetId() &&
-		    stripDetId!=0 && stripDetId->partnerDetId()==dIdOK.rawId())
-		  newhit = false;
-	      //grouped, splitting
-	      if (UseGrouped && UseSplitting)
-		newhit = true;
-	    }
-	    if (newhit) {
-	      tphits.push_back(*TPhit);
-	    }
-	    delete stripDetId;
-	  }
-	  else if (detector == DetId::Muon) {
-	    n_muon_simhits++;
-	    
-	    // discard BAD CSC chambers (ME4/2) from hit counting
-	    if (dId.subdetId() == MuonSubdetId::CSC) {
-              if (csctruth.cscBadChambers->isInBadChamber(CSCDetId(dId))) {
```

Comment:
Hi Chris,

sorry I had forgot...you are right. The functionality is lost from the time of the TrackingParticle redesign about one year a half ago.
Recovering it would require some non-trivial code that I do not think it is worthwhile right now.
cheers

 Giovanni


On Jan 16, 2015, at 9:11 PM, Chris Jones wrote:


In SimMuon/MCTruth/src/MuonAssociatorByHits.cc<https://github.com/cms-sw/cmssw/pull/7175#discussion-diff-23106047>:

> -               newhit = false;
> -           //grouped, splitting
> -           if (UseGrouped && UseSplitting)
> -             newhit = true;
> -         }
> -         if (newhit) {
> -           tphits.push_back(*TPhit);
> -         }
> -         delete stripDetId;
> -       }
> -       else if (detector == DetId::Muon) {
> -         n_muon_simhits++;
> -
> -         // discard BAD CSC chambers (ME4/2) from hit counting
> -         if (dId.subdetId() == MuonSubdetId::CSC) {
> -              if (csctruth.cscBadChambers->isInBadChamber(CSCDetId(dId))) {


This is the only place cscBadChambers was ever accessed and it is commented out.

—
Reply to this email directly or view it on GitHub<https://github.com/cms-sw/cmssw/pull/7175/files#r23106047>.